core/arch/arm/kernel/boot.c

```diff
@@ -653,6 +653,7 @@ static int add_dt_overlay_fragment(struct dt_descriptor *dt, int ioffs)
 
 static int init_dt_overlay(struct dt_descriptor *dt, int __maybe_unused dt_size)
 {
+#if !defined(CFG_EXTERNAL_DTB_OVERLAY_ERASE_AT_BOOT)
```

Comment:
This flag doesn't seem to erase anything, it seems to affect if something is done or not though.
The config flag should have a name which describes what what it does, preferably without negative logic.
Please add some description of this flag in `mk/config.mk`.
If possible favor `if (IS_ENABLED(...))` over `#if ...`.

Reply:
Agreed, is CFG_SKIP_OVERLAY_REUSE more understandable to you ?

Reply:
What would the REUSE part stand for here? (From an OP-TEE point of view)

Reply:
Hum, this is indeed missleading. Currently, if an overlay is given in FDT_ADDR and CFG_CTB_OVERLAY is set, then the DTB overlay will be reused. I guess CFG_IGNORE_BOOT_OVERLAY might be more accurate in this case.

Reply:
But as you said, this is using negative logic so CFG_USE_BOOT_OVERLAY should be more appropriate.

Reply:
Wouldn't it work to just set `CFG_EXTERNAL_DTB_OVERLAY=n`? Or there's something that needs to be done still?

Reply:
Currently, with `CFG_EXTERNAL_DTB_OVERLAY=y`, OP-TEE uses `FDT_ADDR` as a DTB overleay memory area:
* if there is already an overlay: new overlays are added.
* if no valid DTB overlay header found: a DTB overlay format is created and overlays added.
`CFG_EXTERNAL_DTB_OVERLAY=n`, `FDT_ADDR` is not accessed as an DTB overlay area.

I  think @clementleger wants `CFG_EXTERNAL_DTB_OVERLAY=y` but always discarding previous content of `FDT_ADDR` area.

Reply:
@etienne-lms: Yes exactly, currently the overlay always grows when resetting because the DRAM keeps its content.
I would like OP-TEE to recreate the dt overlay at each boot to avoid that.

Reply:
Thanks for the clarification. To me it seems that the DT configuration is getting messier and messier. I think we're lacking a list of the different use cases and which flags used to implement them in each case. As it looks now a small change may have unexpected consequences for a missed use case.

Reply:
@jenswi-linaro I agree, I think such a list would be welcome in `mk/config.mk` and also all combinations that are not explicitly useful should cause an error.

Reply:
After looking at config.mk, I read the following:

```
# Device Tree Overlay support.
# This define enables support for an OP-TEE provided DTB overlay.
# One of two modes is supported in this case:
# 1. Append OP-TEE nodes to an existing DTB overlay located at CFG_DT_ADDR or
#    passed in arg2
# 2. Generate a new DTB overlay at CFG_DT_ADDR
# A subsequent boot stage must then merge the generated overlay DTB into a main
# DTB using the standard fdt_overlay_apply() method.
CFG_EXTERNAL_DTB_OVERLAY ?= n
```

This seems to imply there is 2 modes available but there is not acutally since the logic only check if there is already something at CFG_DT_ADDR.  So I guess a new config variable is actually needed for my use-case (which should be covered by mode 2 described in the previous comment). IMHO, fixing the comment in config.mk and adding a new config variable is needed to really handle both modes.

Reply:
Should I continue adding this feature and adding a comment for it in `mk/config.mk` ?

Reply:
`CFG_EXTERNAL_DTB_OVERLAY` has some legacy in optee_os. Maybe we could keep it for what it describes and introduce a new alternative `CFG_GENERATE_DTB_OVERLAY` which could be described as:
```
# Device Tree Overlay support.
# CFG_EXTERNAL_DTB_OVERLAY allows to append a DTB overlay into an existing
# external DTB. The overlay is created when no valid DTB overlay is found.
# CFG_GENERATE_DTB_OVERLAY allows to create a DTB overlay at external
# DTB location.
# External DTB location (pysical address) is provided either by boot
# argument arg2 or from CFG_DT_ADDR if defined.
# A subsequent boot stage can then merge the generated overlay DTB into a main
# DTB using the standard fdt_overlay_apply() method.
CFG_EXTERNAL_DTB_OVERLAY ?= n
CFG_GENERATE_DTB_OVERLAY ?= n

ifeq (y-y,$(CFG_EXTERNAL_DTB_OVERLAY)-$(CFG_GENERATE_DTB_OVERLAY))
$(error CFG_EXTERNAL_DTB_OVERLAY and CFG_GENERATE_DTB_OVERLAY are exclusive)
endif

Reply:
Looks pretty clear to me and clearly describe the goal of each options ! Thanks for your comment.